### PR TITLE
Require PHP files to only contain PHP

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: tests/cases/php/closing-tag-end [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---DESCRIPTION---
+No closing PHP tags at the end of files
+---CONTENTS---
+<?php
+
+declare(strict_types=1);
+
+$foo = 'bar';
+
+?>
```
Comment (stephenwf) [inline tests/cases/php/closing-tag-end]: Will it simply error if there is text/html after a closing tag?

File: tests/cases/php/closing-tag [hunk 2]
```diff
@@ -9,11 +9,10 @@ $foo = 'bar';
 
 ?>
 
----FIXED---
 <?php
```
Comment (stephenwf) [inline tests/cases/php/closing-tag]: Strange that `?> ... <?php` isn't automatically fixed

File: tests/cases/php/bom [hunk 3]
```diff
@@ -9,4 +9,6 @@ declare(strict_types=1);
 
 ---MESSAGES---
 1:1 Generic.Files.ByteOrderMark.Found
+1:1 Generic.Files.InlineHTML.Found
+1:2 Generic.PHP.CharacterBeforePHPOpeningTag.Found
```
[the remark this answers is not in the document]
Comment (thewilkybarkid) [reply tests/cases/php/bom]: Opened https://github.com/squizlabs/PHP_CodeSniffer/pull/2183.

File: tests/cases/php/closing-tag (hunk 2 above)
Comment (thewilkybarkid) [reply tests/cases/php/closing-tag]: Could in this case as it's just whitespace, but probably a pretty-edge case!

File: tests/cases/php/closing-tag-end (hunk 1 above)
Comment (thewilkybarkid) [reply tests/cases/php/closing-tag-end]: Yep. (`closing-tag` covers that. These aren't meant to be exhaustive unit tests.)

File: tests/cases/php/closing-tag (hunk 2 above)
Comment (stephenwf) [reply tests/cases/php/closing-tag]: very true!
